datafusion/functions/src/datetime/to_char.rs

```diff
@@ -171,7 +171,11 @@ fn _to_char_scalar(
     // of the implementation in arrow-rs we need to convert it to an array
     let data_type = &expression.data_type();
     let is_scalar_expression = matches!(&expression, ColumnarValue::Scalar(_));
-    let array = expression.into_array(1)?;
+    let array_from_expr = expression.into_array(1)?;
```

Comment:
Note for Reviewer:
We can't directly return null when we see that the format is `None` because `format_options` (passed to `ArrayFormatter::try_new`) allow specifying the string to show for null & we need to respect this configuration option.

Reply:
While it is true that `ArrayFormatter::try_new` allows specifying the string to use for null values, I don't think that functionality is exposed via `to_char`

Thus I think this actually should simply return a new `StringArray` of all null values (which confusingly is different than `NullArray`)

So in this case I think if the format is `None` the code should return a null string value (namely `ColumnarValue::Scalar(ScalarValue::Utf8(None))`)

Reply:
Implemented this change. Slight difference though. See https://github.com/apache/arrow-datafusion/pull/9689/files#r1532580061 .